aws/rust-runtime/aws-config/src/profile/app_name.rs

```diff
@@ -0,0 +1,184 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0.
+ */
+
+//! Load an app name from an AWS profile
+
+use crate::provider_config::ProviderConfig;
+use aws_sdk_sts::AppName;
+use aws_types::os_shim_internal::{Env, Fs};
+
+/// Loads an app name from a profile file
+///
+/// This provider will attempt to shared AWS shared configuration and then read the
+/// `sdk-ua-app-id` property from the active profile.
+///
+/// # Examples
+///
+/// **Loads "my-app" as the app name**
+/// ```ini
+/// [default]
+/// sdk-ua-app-id = my-app
```

Comment:
AFAICT, this isn't implemented in other SDKs yet. I'm not aware of any other kebab case properties, so we may want to try to get this changed.

Reply:
It looks like this has been implemented for the Javascript V3 SDK: https://docs.aws.amazon.com/AWSJavaScriptSDK/v3/latest/modules/_aws_sdk_util_user_agent_node.html